src/functions.php

```diff
@@ -306,6 +306,116 @@ function ssh(
     bool $notify = null,
     float $timeout = null,
 ): Process {
+    $ssh = ssh_configuration($host, $user, $sshOptions);
+
+    if ($path) {
+        $command = sprintf('cd %s && %s', $path, $command);
+    }
+
+    return run(
+        $ssh->getExecuteCommand($command),
+        environment: [],
+        tty: false,
+        pty: false,
+        timeout: $timeout,
+        quiet: $quiet,
+        allowFailure: $allowFailure,
+        notify: $notify
+    );
+}
+
+/**
+ * This function is considered experimental and may change in the future.
+ *
+ * @param array{
+ *     'port'?: int,
+ *     'path_private_key'?: string,
+ *     'jump_host'?: string,
+ *     'multiplexing_control_path'?: string,
+ *     'multiplexing_control_persist'?: string,
+ *     'enable_strict_check'?: bool,
+ *     'password_authentication'?: bool,
+ * } $sshOptions
+ */
+function ssh_upload(
+    string $sourcePath,
+    string $destinationPath,
+    string $host,
+    string $user,
+    array $sshOptions = [],
+    bool $quiet = null,
+    bool $allowFailure = null,
+    bool $notify = null,
+    float $timeout = null,
+): Process {
+    $ssh = ssh_configuration($host, $user, $sshOptions);
+
+    return run(
+        $ssh->getUploadCommand($sourcePath, $destinationPath),
+        environment: [],
+        tty: false,
+        pty: false,
+        timeout: $timeout,
+        quiet: $quiet,
+        allowFailure: $allowFailure,
+        notify: $notify
+    );
+}
+
+/**
+ * This function is considered experimental and may change in the future.
+ *
+ * @param array{
+ *     'port'?: int,
+ *     'path_private_key'?: string,
+ *     'jump_host'?: string,
+ *     'multiplexing_control_path'?: string,
+ *     'multiplexing_control_persist'?: string,
+ *     'enable_strict_check'?: bool,
+ *     'password_authentication'?: bool,
+ * } $sshOptions
+ */
+function ssh_download(
+    string $sourcePath,
+    string $destinationPath,
+    string $host,
+    string $user,
+    array $sshOptions = [],
+    bool $quiet = null,
+    bool $allowFailure = null,
+    bool $notify = null,
+    float $timeout = null,
+): Process {
+    $ssh = ssh_configuration($host, $user, $sshOptions);
+
+    return run(
+        $ssh->getDownloadCommand($sourcePath, $destinationPath),
+        environment: [],
+        tty: false,
+        pty: false,
+        timeout: $timeout,
+        quiet: $quiet,
+        allowFailure: $allowFailure,
+        notify: $notify
+    );
+}
+
+/**
+ * @param array{
+ *     'port'?: int,
+ *     'path_private_key'?: string,
+ *     'jump_host'?: string,
+ *     'multiplexing_control_path'?: string,
+ *     'multiplexing_control_persist'?: string,
+ *     'enable_strict_check'?: bool,
+ *     'password_authentication'?: bool,
+ * } $sshOptions
+ */
+function ssh_configuration(
```

Comment:
```suggestion
function ssh(
```

The current `ssh()` function could be renamed to `ssh_run()` then. This would make for a rather clear API.

Of course, this would be breaking but the functions are marked as experimental anyways.